ClassesObjects/Lab/VehicleCatalog/Classes/Vehicle.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+
+namespace VehicleCatalog
+{
+    public abstract class Vehicle : IVehicle
+    {
+        private string brand;
+        private string model;
+        private string type;
+
+        public Vehicle(string type, string brand, string model)
+        {
+            this.Brand = brand;
+            this.Model = model;
+            this.Type = type;
+        }
+
+        public string Brand
+        {
+            get => brand;
+            set
+            {
+                if (IsValidBrand(value))
+                {
+                    brand = value;
+                }
+                else
+                {
+                    throw new ArgumentException("Invalid brand name!");
+                }
+            }
+        }
+        public string Model
+        {
+            get => model;
+            set
+            {
+                if (IsValidModel(value))
+                {
+                    model = value;
+                }
+                else
+                {
+                    throw new ArgumentException("Invalid model name!");
+                }
+            }
+        }
+
+        public string Type {
+            get => type;
+            set
+            {
+                type= value;
+            }
+        }
+        public bool IsValidBrand(string brand)
+        {
+            bool isValid = brand.Length >= 2 && brand.Length <= 15;
+
+            foreach (char c in brand)
+            {
+                if (!Char.IsLetter(c))
+                {
+                    isValid = false;
```

Comment:
You can break here. Also I noticed that you have almost the same foreach in one of the derived classes. Can the common logic be extracted?

Reply:
And in IsValidModel as well